### PR TITLE
ECP Challenge Problem

### PR DESCRIPTION
**Pull request by psakievich**
This PR is to track setting up the ECP challenge problem case. Need to accomplish

- [x] Multiple turbines in the driver input deck
- [x] Removal of unnecessary duplicate files
- [x] Scripted setup of the multiple openfast precursor runs
- [x] Directory management

### CODE REVIEW

File: 4_turb_abl_fsi/template_files/static_box.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+4
```
Comment (ashesh2512) [inline 4_turb_abl_fsi/template_files/static_box.txt]: for a 20B grid - 

4
1
500.0 500.0 0.0 4500.0 4500.0 300.0
1
500.0 500.0 0.0 4500.0 4500.0 300.0
1
500.0 500.0 0.0 4500.0 4500.0 300.0
1
500.0 500.0 0.0 4500.0 4500.0 300.0

File: 4_turb_abl_fsi/template_files/nrel5mw_nalu.yaml [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+      compute_vorticity: yes
+
+  restart:
+    restart_data_base_name: rst01/nrel5mw.rst
+    restart_frequency: 1440
+    restart_start: 0
+
+  output:
+    output_data_base_name: out01/nrel5mw.e
+    output_frequency: 200
```
Comment (ashesh2512) [inline 4_turb_abl_fsi/template_files/nrel5mw_nalu.yaml]: `360`

File: 4_turb_abl_fsi/template_files/static_files/nrel5mw.yaml [hunk 3]
```diff
@@ -0,0 +1,136 @@
+# this may be excessive use of anchors but if we change
+# names and such we can do it all here
+turbine_info: 
+  turb0:
```
Comment (ashesh2512) [inline 4_turb_abl_fsi/template_files/static_files/nrel5mw.yaml:4]: we need something for the mesh tag offset - https://github.com/Exawind/exawind-cases/blob/main/1_turb_abl_fsi/template_files/nrel5mw_nalu.yaml#L183

`mesh_tag_offset: &tag_0 0` ?

File: 4_turb_abl_fsi/template_files/static_files/nrel5mw.yaml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    turbine_hub_pos: &hub_0 [795, 800, 90]
+    turbine_base_pos: &tower_0 [800, 800, 0]
+    openfast_restart: &openfast_rst0 "nrel5mw.5760"
+    force: &forces0 "forces_T0_01.dat"
+    output: &output0 "out01/turb0.e"
+    restart: &restart0 "rst01/turb0.rst"
+    logfile: &log0 "turb_0.log"
+  turb1:
```
Comment (ashesh2512) [inline 4_turb_abl_fsi/template_files/static_files/nrel5mw.yaml:13]: `mesh_tag_offset: &tag_1 1`?

File: 4_turb_abl_fsi/template_files/static_files/nrel5mw.yaml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    turbine_hub_pos: &hub_1 [1595, 1000, 90]
+    turbine_base_pos: &tower_1 [1600, 1000, 0]
+    openfast_restart: &openfast_rst1 "nrel5mw.T1.5760"
+    force: &forces1 "forces_T1_01.dat"
+    output: &output1 "out01/turb1.e"
+    restart: &restart1 "rst01/turb1.rst"
+    logfile: &log1 "turb_1.log"
+  turb2:
+    turbine_hub_pos: &hub_2 [995, 1600, 90]
```
Comment (ashesh2512) [inline 4_turb_abl_fsi/template_files/static_files/nrel5mw.yaml:23]: `mesh_tag_offset: &tag_2 2` ?

File: 4_turb_abl_fsi/template_files/static_files/nrel5mw.yaml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    turbine_base_pos: &tower_2 [1000, 1600, 0]
+    openfast_restart: &openfast_rst2 "nrel5mw.T2.5760"
+    force: &forces2 "forces_T2_01.dat"
+    output: &output2 "out01/turb2.e"
+    restart: &restart2 "rst01/turb2.rst"
+    logfile: &log2 "turb_2.log"
+  turb3:
+    turbine_hub_pos: &hub_3 [1795, 1800, 90]
```
Comment (ashesh2512) [inline 4_turb_abl_fsi/template_files/static_files/nrel5mw.yaml:32]: `mesh_tag_offset: &tag_3 3` ?

**Comment by psakievich** [comment]
Making good progress on this. I need to fix something in the yaml replacement code to allow us to replace a sequence of values (`displacements`). 

Once that and the `tag offsets` noted above are done I think we just need to get the `static_box` and submission params (number of nodes, ranks for nalu vs amr, etc) hammered out and we should be ready to test 🤞🏻

**Comment by ashesh2512** [comment]
@psakievich How are the darwin templates used? Is that for a specific machine?

**Comment by psakievich** [comment]
@ashesh2512 those are so I can run the process on my laptop to ensure that the openfast files are generated properly and the nalu-wind instances are parsed correctly. It will fail when trying to read meshes, but that is the expected behavior.

**Comment by psakievich** [comment]
I think we also need to update this to IDDES after reading the challenge problem language. Do we want to try running this as is first before doing IDDES?  Thoughts?

**Comment by ashesh2512** [comment]
Are you referring to the hybrid RANS/LES terminology? That is referring to RANS in nalu-wind, and LES in amr-wind. I don't believe we need IDDES. In the scenario that we do, let's try to make this work before introducing IDDES. I still don't think we do though.

Edit: Confirmed with Mike Sprague, IDDES is not needed.

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**